src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/Intellisense.cs

```diff
@@ -41,39 +41,21 @@ public IIntellisenseResult Suggest(IntellisenseContext context, TexlBinding bind
 
             IEnumerable<TexlFunction> allFunctionOverloads = null;
 
-            // TODO: Hoist scenario tracking out of language module.
-            // Guid suggestScenarioGuid = Common.Telemetry.Log.Instance.StartScenario("IntellisenseSuggest");
-
-            try
+            if (!TryInitializeIntellisenseContext(context, binding, formula, out var intellisenseData))
             {
-                if (!TryInitializeIntellisenseContext(context, binding, formula, out var intellisenseData))
-                {
-                    allFunctionOverloads = GetFunctionOverloads(intellisenseData.Binding.NameResolver, intellisenseData.CurFunc);
-                    return new IntellisenseResult(new DefaultIntellisenseData(), new List<IntellisenseSuggestion>(), allFunctionOverloads);
-                }
+                allFunctionOverloads = GetFunctionOverloads(intellisenseData.Binding.NameResolver, intellisenseData.CurFunc);
+                return new IntellisenseResult(new DefaultIntellisenseData(), new List<IntellisenseSuggestion>(), allFunctionOverloads);
+            }
 
-                foreach (var handler in _suggestionHandlers)
+            foreach (var handler in _suggestionHandlers)
+            {
+                if (handler.Run(intellisenseData))
                 {
-                    if (handler.Run(intellisenseData))
-                    {
-                        break;
-                    }
+                    break;
                 }
-
-                return Finalize(context, intellisenseData, formula.Loc);
-            }
-            catch (Exception ex)
```

Comment:
>catch [](http://example.com/codeflow?start=12&length=5)

removing this is great - we just need to double check PA and ensure PA has a top-level catcher.

Reply:
I have unrelated compile errors for now
Will try again tomorrow